packages/flutter/lib/src/material/search_anchor.dart

```diff
@@ -682,11 +682,19 @@ class _ViewContentState extends State<_ViewContent> {
     super.initState();
     _viewRect = widget.viewRect;
     _controller = widget.searchController;
+    _controller.addListener(updateSuggestions);
```

Comment:
I think the `updateSuggestions` call can be removed from `defaultTrailing` and `defaultLeading` in the `onPressed` since this change will call `updateSuggestions` when the controller changes its value.